feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/quicksettings/ui/QuickSettingsComponents.kt

```diff
@@ -482,55 +483,58 @@ fun QuickSettingCarouselButton(
  */
 @OptIn(ExperimentalMaterial3ExpressiveApi::class)
 @Composable
-fun QuickSettingCarouselButton(
+fun QuickSettingToggleButton(
     onClick: () -> Unit,
     text: String,
     accessibilityText: String,
     painter: Painter,
     modifier: Modifier = Modifier,
     isHighlighted: Boolean = false,
-    enabled: Boolean = true // todo: handle hidden disabled or visible disabled
+    enabled: Boolean = true
 ) {
-    // todo(kc): better sizing
-    val iconSize = dimensionResource(id = R.dimen.quick_settings_ui_item_icon_size)
+    val buttonSize = IconButtonDefaults.mediumContainerSize(
+        IconButtonDefaults.IconButtonWidthOption.Narrow
+    )
 
     Column(
+        modifier = modifier.width(width = buttonSize.width),
         horizontalAlignment = Alignment.CenterHorizontally,
         verticalArrangement = Arrangement.Center
     ) {
-        ToggleButton(
+        FilledIconToggleButton(
             checked = isHighlighted,
             enabled = enabled,
             onCheckedChange = { _ -> onClick() },
             // 1. Size updated to width 48.dp and height 56.dp
-            modifier = modifier.size(width = 96.dp, height = 112.dp),
-            shapes = ToggleButtonDefaults.shapes()
-                .copy(
-                    checkedShape = RoundedCornerShape(percent = 50),
-                    shape = RoundedCornerShape(percent = 30)
-                ),
-
-            // fixme(kc): the colors of disabled buttons here appear invisible when light mode is enabled
-            colors = ToggleButtonDefaults.toggleButtonColors(
-                containerColor = Color.White.copy(alpha = 0.16F),
-                contentColor = MaterialTheme.colorScheme.secondaryFixed,
-                checkedContainerColor = MaterialTheme.colorScheme.primaryFixedDim,
-                checkedContentColor = MaterialTheme.colorScheme.onPrimaryFixed
-            )
+            modifier = Modifier
+                .minimumInteractiveComponentSize()
+                .size(buttonSize),
+            shapes = IconButtonDefaults.toggleableShapes().let {
+                // invert the selected shapes
+                it.copy(
+                    shape = it.checkedShape,
+                    checkedShape = it.shape
+                )
+            },
```

Comment:
It feels odd that this is inverted. We should go back to the design and check why.